File: python/剑指offer/面试题22 链表中倒数第k个节点.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        p1,p2 = head,head
        while k:
            p2 = p2.next
            k-=1
        while p2:
            p2 = p2.next
            p1 = p1.next
        return p1

File: go/剑指offer/面试题22. 链表中倒数第k个节点.go
package main

func getKthFromEnd(head *ListNode, k int) *ListNode {
	p1, p2 := head, head
	for k > 0 {
		p2 = p2.Next
		k -= 1
	}
	for p2 != nil {
		p2 = p2.Next
		p1 = p1.Next
	}
	return p1
}
